lib/api/src/js/exports.rs

```diff
@@ -141,51 +149,61 @@ impl Exports {
     /// Get an export as a `TypedFunction`.
     pub fn get_native_function<Args, Rets>(
         &self,
+        ctx: &impl AsContextRef,
         name: &str,
     ) -> Result<TypedFunction<Args, Rets>, ExportError>
     where
         Args: WasmTypeList,
         Rets: WasmTypeList,
     {
-        self.get_typed_function(name)
+        self.get_typed_function(ctx, name)
     }
 
     /// Get an export as a `TypedFunction`.
     pub fn get_typed_function<Args, Rets>(
         &self,
+        ctx: &impl AsContextRef,
         name: &str,
     ) -> Result<TypedFunction<Args, Rets>, ExportError>
     where
         Args: WasmTypeList,
         Rets: WasmTypeList,
     {
-        self.get_function(name)?
-            .native()
+        self.get_function(&ctx.as_context_ref(), name)?
+            .native(ctx)
             .map_err(|_| ExportError::IncompatibleType)
     }
 
     /// Hack to get this working with nativefunc too
-    pub fn get_with_generics<'a, T, Args, Rets>(&'a self, name: &str) -> Result<T, ExportError>
+    pub fn get_with_generics<'a, T, Args, Rets>(
+        &'a self,
+        ctx: &impl AsContextRef,
+        name: &str,
+    ) -> Result<T, ExportError>
     where
         Args: WasmTypeList,
         Rets: WasmTypeList,
         T: ExportableWithGenerics<'a, Args, Rets>,
     {
         match self.map.get(name) {
             None => Err(ExportError::Missing(name.to_string())),
-            Some(extern_) => T::get_self_from_extern_with_generics(extern_),
+            Some(extern_) => T::get_self_from_extern_with_generics(ctx, extern_),
         }
     }
 
     /// Like `get_with_generics` but with a WeakReference to the `InstanceRef` internally.
-    /// This is useful for passing data into `WasmerEnv`, for example.
-    pub fn get_with_generics_weak<'a, T, Args, Rets>(&'a self, name: &str) -> Result<T, ExportError>
+    /// This is useful for passing data into Context data, for example.
+    pub fn get_with_generics_weak<'a, T, Args, Rets>(
+        &'a self,
+        ctx: &impl AsContextRef,
+        name: &str,
+    ) -> Result<T, ExportError>
     where
         Args: WasmTypeList,
         Rets: WasmTypeList,
         T: ExportableWithGenerics<'a, Args, Rets>,
     {
-        let out: T = self.get_with_generics(name)?;
+        let out: T = self.get_with_generics(ctx, name)?;
         Ok(out)
     }
```

Comment:
I believe this doesn't exist any longer

Reply:
Ok, removing...

Reply:
Looks like it exists on the `sys` side, so keeping it.